src/components/PupilViews/PupilQuiz/PupilQuiz.view.tsx

```diff
@@ -1,18 +1,135 @@
 import {
   OakCloudinaryConfigProvider,
-  OakFlex,
+  OakBackLink,
+  OakLessonBottomNav,
+  OakLessonLayout,
+  OakLessonTopNav,
+  OakPrimaryButton,
+  OakQuizCounter,
+  OakSpan,
 } from "@oaknational/oak-components";
 
 import {
   QuestionsArray,
   QuizEngineProvider,
+  useQuizEngineContext,
 } from "@/components/PupilComponents/QuizEngineProvider";
 import { QuizRenderer } from "@/components/PupilComponents/QuizRenderer";
+import { useLessonEngineContext } from "@/components/PupilComponents/LessonEngineProvider";
+import { pickFeedBackComponent } from "@/components/PupilComponents/QuizUtils/pickFeedback";
 
 type PupilViewsQuizProps = {
   questionsArray: QuestionsArray;
 };
 
+type QuizSection = "starter-quiz" | "exit-quiz";
+
+const isQuizSection = (section: string): section is QuizSection => {
+  return ["starter-quiz", "exit-quiz"].includes(section);
+};
+
+const QuizInner = () => {
+  const { currentSection, updateCurrentSection } = useLessonEngineContext();
+  const quizEngineContext = useQuizEngineContext();
+
+  if (!isQuizSection(currentSection)) {
+    return null;
+  }
+
+  const {
+    currentQuestionData,
+    currentQuestionIndex,
+    questionState,
+    handleNextQuestion,
+    numQuestions,
+  } = quizEngineContext;
+
+  const isFeedbackMode =
+    questionState[currentQuestionIndex]?.mode === "feedback";
+  const isCorrect = questionState[currentQuestionIndex]?.grade === 1;
+  const isAlmostCorrect = questionState[currentQuestionIndex]?.grade === 0.5;
+  const formId = "quiz-form";
+
+  const bottomNavSlot = (
+    <OakLessonBottomNav
+      feedback={
+        isFeedbackMode
+          ? isCorrect
+            ? "correct"
+            : isAlmostCorrect
+              ? "partially-correct"
+              : "incorrect"
+          : null
+      }
```

Comment:
🏄🏼 fwiw i'd have extracted this into a little function `pickFeedback` so that most of the logic is outside the JSX.

```jsx
<OakLessonBottomNav
  feedback={isFeedbackMode ? pickFeedback(grade) : null}
  ...
</OakLessonBottomNav>
```

```typescript
function pickFeedback(grade: number) {
  switch (grade) {
    case 1:
      return 'correct';
    case 0:
      return 'incorrect';
    case 0.5:
      return 'partially-correct';
    default:
      return null;
  }
}
```

when i'm reading a large block of JSX it's really helpful if I can quickly work out the intent from names of things without having to follow the implementation all the way through.